core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregationBuilder.java

```diff
@@ -88,40 +121,61 @@ protected static long parseStringOffset(String offset) {
                 .millis();
     }
 
-    public DateHistogramInterval dateHistogramInterval() {
-        return dateHistogramInterval;
+    public ExtendedBounds extendedBounds() {
+        return extendedBounds;
     }
 
-    @Override
-    protected DateHistogramAggregatorFactory innerBuild(AggregationContext context, ValuesSourceConfig<Numeric> config,
-            AggregatorFactory<?> parent, Builder subFactoriesBuilder) throws IOException {
-        return new DateHistogramAggregatorFactory(name, type, config, interval, dateHistogramInterval, offset, order, keyed, minDocCount,
-                extendedBounds, context, parent, subFactoriesBuilder, metaData);
+    public DateHistogramAggregationBuilder extendedBounds(ExtendedBounds extendedBounds) {
+        if (extendedBounds == null) {
+            throw new IllegalArgumentException("[extendedBounds] must not be null: [" + name + "]");
+        }
+        this.extendedBounds = extendedBounds;
+        return this;
     }
 
     @Override
-    public String getWriteableName() {
-        return NAME;
-    }
+    protected XContentBuilder doXContentBody(XContentBuilder builder, Params params) throws IOException {
 
-    @Override
-    protected XContentBuilder doXContentInterval(XContentBuilder builder, Params params) throws IOException {
         if (dateHistogramInterval == null) {
-            super.doXContentInterval(builder, params);
+            builder.field(Histogram.INTERVAL_FIELD.getPreferredName(), interval);
         } else {
-            builder.value(dateHistogramInterval.toString());
+            builder.field(Histogram.INTERVAL_FIELD.getPreferredName(), dateHistogramInterval.toString());
         }
+        builder.field(Histogram.OFFSET_FIELD.getPreferredName(), offset);
+
+        super.doXContentBody(builder, params);
+
+        if (extendedBounds != null) {
+            extendedBounds.toXContent(builder, params);
+        }
+
         return builder;
     }
 
     @Override
+    public String getWriteableName() {
+        return InternalDateHistogram.TYPE.name();
```

Comment:
I think `InternalDateHistogram.NAME` would be a bit better. Now that type isn't used for serialization I'd like to remove it entirely one day.